src/ts/utils/react-hotkeys.ts

```diff
@@ -0,0 +1,126 @@
+import {KEY_TO_CODE} from './keyboard'
+
+const invertObject = <K, V>(obj: {
+    // @ts-ignore
+    [key: K]: V
+}) => {
+    const inverted = {}
+    for (let [key, value] of Object.entries(obj)) {
+        // Fallback to taking the first one, if multiple values conflict
+        // @ts-ignore
+        inverted[value] = inverted[value] || key
+    }
+    return inverted
+}
+
+/**
+ * React Hotkeys treats j/J as different keys, which is buggy when detecting key Combinations.
+ * For example, with this sequence:
+ *
+ * [shift down, j down, shift up, j up]
+ *
+ *
+ *        ---------Shift+J-------
+ *        |                     |
+ * [Shift+J down, J down, Shift+J up, j up]
+ *                   |                  |
+ *                   ----Not Released----
+ *
+ * React Hotkeys thinks that the "J" key is still down, which breaks all hotkeys until you
+ * Un-focus the browser window (resetting the key history).
+ *
+ * To workaround this issue, we normalize j/J to be the "same key" using the keycode,
+ * That way, a "J" keydown is stored as a "j" keydown in the key history, so a "j" keyup
+ * will "release" it.
+ *
+ * Related issue: https://github.com/greena13/react-hotkeys/issues/249
+ *
+ * React Hotkeys doesn't have an active maintainer, so it's harder to fix upstream.
```

Comment:
should we consider switching to something else?

Reply:
I think it's worth researching, I filed https://github.com/roam-unofficial/roam-toolkit/issues/68